docs/features/software-templates/writing-custom-actions.md

```diff
@@ -107,6 +107,17 @@ export const createNewFileAction = () => {
 };
 ```
 
+#### A note on naming conventions
+
+Try to keep names consistent for both your own custom actions, and any actions contributed to open source. We've found that a separation of `:` and using a verb as the last part of the name works well.
+We follow `provider:entity:verb` or as close to this as possible for our built in actions. For example, `github:actions:create` or `github:repo:create`.
+
+Also feel free to use your company name to namespace them if you prefer too, for example `acme:file:create`.
+
+Prefer to use `camelCase` over `snake-case` for these actions if possible, which leads to better reading and writing of template entity definitions.
```

Comment:
I was under the impression that it's the other way around tbh 😅. The only ones we camel case are where we need to match the provider IDs for bitbucket right?

Reply:
The reason I thought of adding this, is because if you have `stepIds` in templates and use them as outputs you have to do:

`${{ steps['my-step'].output.something }}` rather than `${{ steps.myStep.output.something }}` and thought that was less of an eyesore and something to know about, and then forcing the action names to be `camelCase` would also reduce the notion that you would do this in the `template.yaml`.

Happy to revert, but thought that maybe it makes sense?

Reply:
Yep no particularly strong opinion on which one we choose tbh. Whatever we do end up picking I'm think it's best if we update all built-in action names to align, with a deprecation path ofc.

Reply:
Will create a ticket to follow up with these renames